sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -620,26 +620,43 @@ object DateTimeUtils {
   // Visible for testing.
   private[sql] val TRUNC_TO_YEAR = 1
   private[sql] val TRUNC_TO_MONTH = 2
-  private[sql] val TRUNC_TO_QUARTER = 3
-  private[sql] val TRUNC_TO_WEEK = 4
-  private[sql] val TRUNC_TO_DAY = 5
-  private[sql] val TRUNC_TO_HOUR = 6
-  private[sql] val TRUNC_TO_MINUTE = 7
-  private[sql] val TRUNC_TO_SECOND = 8
+  private[sql] val TRUNC_TO_DECADE = 3
```

Comment:
Is it possibly important at all to keep the existing numbering? 
Or if not, should this be sorted by length of time?

Reply:
I have sorted by lengths in 2 separate ranges for `DATE`/`TIMESTAMP` and for `TIMESTAMP`.